service/bin/geopmd

```diff
@@ -0,0 +1,103 @@
+import geopmpy.pio
+
+class GEOPMService(object):
+    __dbus_xml__ = """
+    <node>
+        <interface name="io.github.geopm0">
+            <method name="AccessList">
+                <arg direction="in" name="group" type="s" />
+                <arg direction="out" name="allowed_signals" type="as" />
+                <arg direction="out" name="allowed_controls" type="as" />
+            </method>
+            <method name="AccessControl">
+                <arg direction="in" name="group" type="s" />
+                <arg direction="in" name="allowed_signals" type="as" />
+                <arg direction="in" name="denied_signals" type="as" />
+                <arg direction="in" name="allowed_controls" type="as" />
+                <arg direction="in" name="denied_controls" type="as" />
+            </method>
+            <method name="AccessDelete">
+                <arg direction="in" name="group" type="s" />
+            </method>
+            <method name="AccessAllowed">
+                <arg direction="out" name="allowed_signals" type="a(si)" />
+                <arg direction="out" name="allowed_controls" type="a(si)" />
+            </method>
+            <method name="Loop">
+                <arg direction="in" name="signals" type="a(sii)" />
+                <arg direction="in" name="controls" type="a(sii)" />
+                <arg direction="in" name="interval" type="d" />
+                <arg direction="out" name="loop_pid" type="i" />
+                <arg direction="out" name="signal_shmem" type="s" />
+                <arg direction="out" name="control_shmem" type="s" />
+                <arg direction="out" name="clock_start" type="(xx)" />
+            </method>
+            <method name="SignalInfo">
+                <arg direction="in" name="query" type="as" />
+                <arg direction="out" name="result" type="a(siii)" />
+            </method>
+            <method name="ControlInfo">
+                 <arg direction="in" name="query" type="as" />
+                 <arg direction="out" name="result" type="as" />
+            </method>
+        </interface>
+    </node>
+    """
+    def __init__(self, pio=geopmpy.pio):
+        self._pio = pio
+
+    def AccessList(self, group):
+        return self._pio.signal_names(), self._pio.control_names()
+
+    def AccessControl(self, group,
+                      allowed_signals, denied_signals,
+                      allowed_controls, denied_controls):
+        raise NotImplementedError
+
+    def AccessDelete(self, group):
+        raise NotImplementedError
+
+    def AccessAllowed(self):
+        names = self._pio.signal_names()
+        signals = [(nn, self._pio.signal_domain_type(nn))
+                   for nn in self._pio.signal_names()]
+        controls = [(nn, self._pio.control_domain_type(nn))
+                    for nn in self._pio.control_names()]
+        return signals, controls
+
+    def Loop(self, signals, controls, interval):
```

Comment:
Maybe `StartXYZLoop`? I guess `XYZ` could be something like `PlatformIO`, `Sampling`, `Control`, `GeopmControl`, etc. Or maybe `StartGeopmController`

Reply:
Decided on a concept of opening and closing a "session".   LMKWYT